src/pages/iou/steps/MoneyRequestAmountPage.js

```diff
@@ -99,9 +101,15 @@ class MoneyRequestAmountPage extends React.Component {
     }
 
     componentDidMount() {
+        const moneyRequestId = `${this.iouType}${this.reportID}`;
+        const shouldReset = this.props.iou.id !== moneyRequestId;
+        if (shouldReset) {
+            IOU.resetMoneyRequestInfo(moneyRequestId);
+        }
+
         if (this.isEditing) {
-            if (_.isEmpty(this.props.iou.participants) || this.props.iou.amount === 0) {
-                Navigation.goBack(ROUTES.getMoneyRequestRoute(this.iouType, this.reportID));
+            if (_.isEmpty(this.props.iou.participants) || this.props.iou.amount === 0 || shouldReset) {
+                Navigation.goBack(ROUTES.getMoneyRequestRoute(this.iouType, this.reportID), shouldReset);
                 return;
             }
         }
```

Comment:
This can be combined into one if condition.

Reply:
You're right. Updated!

Reply:
Cool, mark this as resolved